abm/server.py

```diff
@@ -145,6 +146,38 @@ def do_get(self, context, ticket: fl.Ticket):
         # return dataset as arrow flight record batches
         return fl.GeneratorStream(schema, batches)
 
+    '''
+    Serve arrow flight do_put requests
+    '''
+    def do_put(self, context, descriptor, reader, writer):
+        asset_name = json.loads(descriptor.command)['asset']
+        schema = json.loads(descriptor.command)['schema']
```

Comment:
No need to run `json.loads(descriptor.command)` twice.
Please check whether the 'asset' and 'schema' fields appear in the map, and return an error if needed